mergify_engine/queue/merge_train.py

```diff
@@ -645,9 +649,29 @@ async def _set_initial_state(
             self.train.repository.log,
             False,
         )
+
+        if queue_rule.config["checks_timeout"] is not None:
+            await self.set_checks_timeout_context(queue_rule.config["checks_timeout"])
```

Comment:
I think we should update the delayed refresh date each time the train car state is updated not only when is created? That was the case in previous revision I think, no?